wpimath/src/main/java/edu/wpi/first/math/geometry/Pose2d.java

```diff
@@ -238,6 +241,18 @@ public Twist2d log(Pose2d end) {
     return new Twist2d(translationPart.getX(), translationPart.getY(), dtheta);
   }
 
+  /**
+   * Returns the nearest Pose2d from a list of poses.
+   *
+   * @param poses The list of poses to find the nearest.
+   * @return The nearest Pose2d form the list.
+   */
+  public Pose2d nearest(List<Pose2d> poses) {
+    return Collections.min(
+        poses,
+        Comparator.comparing(other -> this.getTranslation().getDistance(other.getTranslation())));
```

Comment:
🤔 what if there are two Pose2ds with equal translation distance but different rotation? I'd expect it to tie break on rotation

Reply:
This is pretty trivial to do- I'll update this later today